src/codesearch.cc

```diff
@@ -206,18 +207,15 @@ class searcher {
 public:
     searcher(const code_searcher *cc,
              const query &q,
+             const intrusive_ptr<IndexKey> index,
```

Comment:
can we call this `key` or something instead of `index`? "index" refers in my mind to the data structure with the suffix arrays, and an "index key" is a thing that one can look up efficiently in it.